codec/src/main/java/io/netty/handler/codec/DateFormatter.java

```diff
@@ -171,7 +171,7 @@ public void reset() {
         month = -1;
         yearFound = false;
         year = -1;
-        cal.set(Calendar.MILLISECOND, 0);
+        cal.clear();
```

Comment:
I wonder if this will have any impact on perf. Under the hood it looks like this iterates over a `17` element array. Can  you profile the change to make sure there is no negative impact?